src/discovery/local-discovery.js

```diff
@@ -233,6 +233,16 @@ export class LocalDiscovery extends TypedEmitter {
         return
       }
     }
+
+    // Bail if the server has already stopped by this point. This should be
+    // rare but can happen during connection swaps if the new connection is
+    // "promoted" after the server's doors have already been closed.
+    if (!this.#server.listening) {
+      this.#log('server stopped, destroying connection %h', remotePublicKey)
+      conn.destroy()
+      return
+    }
```

Comment:
I'm confident[^hubris] in my understanding of the problem...less so in my solution. Do we want to do this if the server is not force-closed, for example?

[^hubris]: this might be hubris